activerecord/test/cases/persistence_test.rb

```diff
@@ -912,6 +922,17 @@ def test_class_level_destroy
     assert_raise(ActiveRecord::RecordNotFound) { Reply.find(should_be_destroyed_reply.id) }
   end
 
+  def test_class_level_destroy_is_affected_by_scoping
+    should_be_destroyed_reply = Reply.create("title" => "hello", "content" => "world")
```

Comment:
`should_be_destroyed_reply` is a misleading name for a reply which you then verify has not been destroyed.